### DISCUSSION TITLE
are prebuilt releases available and instructions to install on current macos ventura 13.4

### DISCUSSION THREAD

**Discussion by myusrn** [neutrinolabs/xrdp#2725] [Q&A] [answered]
When i look through the readme.md and wiki content it appears this work is being distributed in the form where you have to sync sources and build yourself.  

Are prebuilt daily or weekly releases available along with instructions to install it on current macos ventura 13.4 environments?

If not are there step by step instructions for citizen developers to follow to sync and build and install on macOS?  when i click the wiki | building from sources | mac os link it takes me to a create a new page | buiding on osx blank result.

**Comment by metalefty** [comment, marked as answer]
Hi, sorry for the dead link.

See here: https://github.com/neutrinolabs/xrdp/wiki/Building-on-OSX-(not-official)

**Reply by myusrn** [reply]
@metalefty thanks for the pointer to the document walking one through how to sync, build and install using current sources.  

It seems like this is the only hope for free remote desktop solution to imac that produces usable performance experience given the "screen sharing" setting coupled with a real vnc viewer client to connect to it from anything that isn't another macos setup performs so badly it suggests apple purposely made it that way.

**Reply by myusrn** [reply]
Hi @metalefty and others.

I followed the instructions in building-on-osx wiki noted above and the build process for xrdp and xorgxrdp worked w/o any errors. I added `/usr/local/sbin` to path and ran `sudo /usr/local/sbin/xrdp` then attempted to connect from windows rdp store app.  

I get a black screen for 5-10 seconds then see an error dialog `connection log | error connecting to sessman on 127.0.0.1:3350 | okay` which i dismiss and am presented the xrdp signin dialog with Xorg and username populated.  I enter password and it tries for 5-10 seconds and it does nothing but bring me back to signin screen.

I then tried logging out of the desktop and using sharing | remote login enabled ssh terminal connection to issue the `sudo /usr/local/sbin/xrdp` command and reattempted the windows rdp store app connection and got same result.

I then tried to enable /Library/LaunchDaemons setup so i could test having xrdp service running w/o any user currently logged in on physical console to see if that produced a different result.  I used the following steps to do that and when i did a port query for tcp/3389 there was nothing listening.  So it seems i'm missing some critical step to run that test.  

`sudo nano /Library/LaunchDaemons/com.neutrinolabs.xrdp.plist`

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
    <key>Label</key>
    <string>com.xrdp</string>
    <key>ProgramArguments</key>
    <array>
        <string>/usr/local/xrdp</string>
    </array>
    <key>RunAtLoad</key>
    <true/>
</dict>
</plist>
```
`sudo chown root:wheel /Library/LaunchDaemons/com.xrdp.plist`

`sudo launchctl load -w /Library/LaunchDaemons/com.xrdp.plist`

Perhaps i have to move /usr/local/sbin/xrdp* files in /usr/sbin in order for it to run as a /Library/LaunchDaemons setup. Attempts to do that using `sudo mv /usr/local/sbin/xrdp* /usr/sbin` produced `mv: fastcopy: open() failed (to): /usr/sbin/<xrdp filename>: Operation not permitted` errors.

I'm trying to rebuild everything to see if i can get a different result using ./configure --disable-ipv6 switch which i found mentioned in issue #1855 as being a way to resolve error dialogs mentioning sesman. While attempting to do so the xorgxrdp make step was now generating the following error.  I executed a `find / -name Xosdefs.h 2>/dev/null` and the file its saying it can't find is showing up in `/opt/X11/include/X11/Xosdefs.h`. So unclear why this is happening or how to fix it.

```
In file included from /opt/X11/include/xorg/xf86str.h:37:
/opt/X11/include/xorg/misc.h:74:10: fatal error: 'X11/Xosdefs.h' file not found
#include <X11/Xosdefs.h>
1 error generated.
```
